### PR TITLE
debug: limit the size of the trace

### PR DESCRIPTION
**Pull request by dnephin**
We've noticed that a trace that is captured over the full duration is too large to open on most machines (it fails to open with an out-of-memory error while trying to read the trace.out). A trace.out captured over just the interval period (30s by default) should be a more than enough time to capture relevant debugging data.

Another option would be to add a new command line flag for this value, but I suspect that extra work is not worth it.

### CODE REVIEW

File: command/debug/debug.go [hunk 1]
```diff
@@ -413,7 +413,7 @@ func (c *cmd) captureLongRunning(ctx context.Context) error {
 
 		g.Go(func() error {
 			// use ctx without a timeout to allow the trace to finish sending
-			return c.captureTrace(ctx, s)
+			return c.captureTrace(ctx, int(c.interval.Seconds()))
```
Comment (freddygv) [inline command/debug/debug.go:416]: Unrelated to this change, but I just noticed that the adjustment in L405-407 isn't doing anything given that there's a `debugMinDuration` set to 10s.

**Review by freddygv — APPROVED** [review]
Seems reasonable to me. 

Are there any docs changes needed?

File: command/debug/debug.go (hunk 1 above)
Comment (dnephin) [reply command/debug/debug.go:416]: Ya, I noticed that as well, but decided to leave it for now to keep the PR focused on this fix. I'm happy to remove those lines in another commit.

**Comment by dnephin** [comment]
> Are there any docs changes needed?

I updated the CLI docs in the latest commit to clarify how trace is captured.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/585596.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 8110ecc93bb665e1d7190c5b5f20a0cca109304e onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 8110ecc93bb665e1d7190c5b5f20a0cca109304e onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/585596)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 8110ecc93bb665e1d7190c5b5f20a0cca109304e onto `release/1.9.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/585596)

**Pull request merged**